### PR TITLE
[REEF-1204] Interface separation between Job and Application Parameters

### PR DESCRIPTION
**Pull request by afchung**
This addressed the issue by
  * Deprecate IJobSubmission and its related classes and interfaces.
  * Added JobRequest, JobParameters, AppParameters, and their related classes.
  * Added methods to submit jobs with the newly introduced classes.

JIRA:
  [REEF-1204](https://issues.apache.org/jira/browse/REEF-1204)

### CODE REVIEW

**Comment by afchung** [comment]
@anupam128 @beysims Please have a look when you have time. Thanks!

File: lang/cs/Org.Apache.REEF.Client/API/AppParameters.cs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+using Org.Apache.REEF.Tang.Interface;
+
+namespace Org.Apache.REEF.Client.API
+{
+    /// <summary>
+    /// The parameters for a REEF application, used to specify application parameters for each REEF application.
+    /// For job parameters which is specified each time for job submissions of the same
+    /// REEF application, see <see cref="JobParameters"/>.
+    /// </summary>
+    public sealed class AppParameters
```
Comment (anupam128) [inline lang/cs/Org.Apache.REEF.Client/API/AppParameters.cs]: Job vs app is not implicitly clear. How can we make this more apparent?

**Comment by anupam128** [comment]
This LGTM. This is an API deprecation so I would like @markusweimer / @jwang98052 / @beysims to take a look as well.

**Comment by markusweimer** [comment]
@shulmanb Do you use any of the affected APIs?

**Comment by afchung** [comment]
@markusweimer ASA uses their own client.

**Comment by markusweimer** [comment]
In that case, I have no issues with merging this.

**Comment by afchung** [comment]
@anupam128 could you merge please? Thanks!

**Comment by anupam128** [comment]
Yes. I will test and merge this change.

**Pull request closed without merging**